src/Categories/Examples/Monad/Duration.agda

```diff
@@ -0,0 +1,99 @@
+{-# OPTIONS --safe --without-K #-}
+module Duration where
+
+open import Level
+
+open import Function using (_$_)
+open import Function.Equality using (Π; _⟨$⟩_; cong)
+
+open import Relation.Binary
+open import Relation.Binary.Indexed.Heterogeneous.Construct.Trivial
+
+open import Data.Product using (Σ; _,_; _×_; proj₁; proj₂)
+open import Algebra using (Monoid; IsMonoid; IsSemigroup; IsMagma)
+
+open import Categories.Category
+open import Categories.Category.Instance.Setoids
+open import Categories.Functor renaming (id to idF)
+open import Categories.NaturalTransformation using (NaturalTransformation)
+open import Categories.Monad using (Monad)
+
+open import Data.Nat.Properties renaming (+-0-monoid to +-0-nat-monoid)
+open import Data.Integer.Properties renaming (+-0-monoid to +-0-integer-monoid)
+open import Data.Rational.Properties renaming (+-0-monoid to +-0-rational-monoid)
+
+open Monoid public
+
+monoid×-₀ : Monoid 0ℓ 0ℓ → Setoid 0ℓ 0ℓ → Setoid 0ℓ 0ℓ
+monoid×-₀ record { Carrier = Carrierₘ ; _≈_ = _≈ₘ_ ; isMonoid = isMonoid } record { Carrier = Carrier ; _≈_ = _≈_ ; isEquivalence = isEquivalence } = record
+  { Carrier = Carrierₘ × Carrier
+  ; _≈_ = λ (x₁ , y₁) (x₂ , y₂) → x₁ ≈ₘ x₂ × y₁ ≈ y₂
+  ; isEquivalence = record
+      { refl = IsEquivalence.refl isEquivalenceₘ , IsEquivalence.refl isEquivalence
+      ; sym = λ (fst , snd) → IsEquivalence.sym isEquivalenceₘ fst , IsEquivalence.sym isEquivalence snd
+      ; trans = λ (fst₁ , snd₁) (fst₂ , snd₂) → IsEquivalence.trans isEquivalenceₘ fst₁ fst₂ , IsEquivalence.trans isEquivalence snd₁ snd₂
+      }
+  } where isEquivalenceₘ = IsMagma.isEquivalence $ IsSemigroup.isMagma $ IsMonoid.isSemigroup isMonoid
+
+monoid×-₁ : ∀ {A B : Setoid 0ℓ 0ℓ} → (m : Monoid 0ℓ 0ℓ) → (f : Setoids 0ℓ 0ℓ [ A , B ]) → Setoids 0ℓ 0ℓ [ monoid×-₀ m A , monoid×-₀ m B ]
+monoid×-₁ _ record { _⟨$⟩_ = f ; cong = cong } = record
+  { _⟨$⟩_ = λ (fst ,  snd) → fst , f snd
+  ; cong = λ (fst , snd) → fst , cong snd
+  }
+
+monoid×-hom : {x y z : Setoid 0ℓ 0ℓ} {m : Monoid zero zero} {x₁ y₁ : Σ (Carrier m) (λ _ → Setoid.Carrier x)}
+    → (f : Π x (indexedSetoid y))
+    → (g : Π y (indexedSetoid z))
+    → (x Setoid.≈ proj₂ x₁) (proj₂ y₁)
+    → (z Setoid.≈ g ⟨$⟩ (f ⟨$⟩ proj₂ x₁)) (g ⟨$⟩ (f ⟨$⟩ proj₂ y₁))
+monoid×-hom record { cong = congᵐ } record { cong = cong } p = cong $ congᵐ p
+
+monoid×-endo : Monoid 0ℓ 0ℓ → Endofunctor (Setoids 0ℓ 0ℓ)
+monoid×-endo m =  record
+  { F₀ = monoid×-₀ m
+  ; F₁ = monoid×-₁ m
+  ; identity = λ x → x
+  ; homomorphism = λ {_} {_} {_} {f} {g} {x₁} {y₁} (fst , snd) → fst , monoid×-hom {_} {_} {_} {m} {x₁} {y₁} f g snd
```

Comment:
If the implicits are needed, the a comment to that effect, please. Otherwise, leave them implicit.